apps/betterangels-backend/shelters/tests/test_queries.py

```diff
@@ -0,0 +1,344 @@
+            website="shelter.com",
+            _quantity=self.shelter_count,
+        )
+
+    def _setup_shelter_related_objects(self) -> None:
+        shelter_related_objects: List[ShelterRelatedObject] = [
+            ShelterRelatedObject(
+                field_name="accessibility", model_name="Accessibility", value=AccessibilityChoices.WHEELCHAIR_ACCESSIBLE
+            ),
+            ShelterRelatedObject(field_name="cities", model_name="City", value=CityChoices.AGOURA_HILLS),
+            ShelterRelatedObject(field_name="demographics", model_name="Demographic", value=DemographicChoices.ALL),
+            ShelterRelatedObject(
+                field_name="entry_requirements", model_name="EntryRequirement", value=EntryRequirementChoices.PHOTO_ID
+            ),
+            ShelterRelatedObject(field_name="funders", model_name="Funder", value=FunderChoices.CITY_OF_LOS_ANGELES),
+            ShelterRelatedObject(
+                field_name="general_services", model_name="GeneralService", value=GeneralServiceChoices.CASE_MANAGEMENT
+            ),
+            ShelterRelatedObject(
+                field_name="health_services", model_name="HealthService", value=HealthServiceChoices.DENTAL
+            ),
+            ShelterRelatedObject(
+                field_name="immediate_needs", model_name="ImmediateNeed", value=ImmediateNeedChoices.CLOTHING
+            ),
+            ShelterRelatedObject(field_name="parking", model_name="Parking", value=ParkingChoices.BICYCLE),
+            ShelterRelatedObject(field_name="pets", model_name="Pet", value=PetChoices.CATS),
+            ShelterRelatedObject(field_name="room_styles", model_name="RoomStyle", value=RoomStyleChoices.CONGREGANT),
+            ShelterRelatedObject(
+                field_name="shelter_programs", model_name="ShelterProgram", value=ShelterProgramChoices.BRIDGE_HOME
+            ),
+            ShelterRelatedObject(field_name="shelter_types", model_name="ShelterType", value=ShelterChoices.BUILDING),
+            ShelterRelatedObject(field_name="spa", model_name="SPA", value=SPAChoices.ONE),
+            ShelterRelatedObject(
+                field_name="special_situation_restrictions",
+                model_name="SpecialSituationRestriction",
+                value=SpecialSituationRestrictionChoices.NONE,
+            ),
+            ShelterRelatedObject(field_name="storage", model_name="Storage", value=StorageChoices.AMNESTY_LOCKERS),
+            ShelterRelatedObject(
+                field_name="training_services", model_name="TrainingService", value=TrainingServiceChoices.JOB_TRAINING
+            ),
+        ]
+
+        for i in shelter_related_objects:
+            model_cls = apps.get_model("shelters", i.model_name)
+            related_object = baker.make(model_cls, name=i.value)
+
+            for shelter in self.shelters:
+                related_manager = getattr(shelter, i.field_name)
+                related_manager.add(related_object)
```

Comment:
might be an opportunity to make a recipe with model bakery

https://model-bakery.readthedocs.io/en/latest/recipes.html

Reply:
Might look something like this

```python
from model_bakery.recipe import Recipe, seq, related
from shelters.models import Shelter, Accessibility, City, Demographic, ContactInfo, ExteriorPhoto, InteriorPhoto

# Related object recipes
accessibility_recipe = Recipe(Accessibility, name="Wheelchair Accessible")
city_recipe = Recipe(City, name="Agoura Hills")
demographic_recipe = Recipe(Demographic, name="All")

# Main Shelter recipe with related objects
shelter_recipe = Recipe(
    Shelter,
    name=seq("Shelter "),  # Sequential naming
    bed_fees="bed fees",
    city_council_district=1,
    curfew="22:00:00",
    demographics_other="demographics other",
    description="description",
    email="shelter@example.com",
    entry_info="entry info",
    funders_other="funders other",
    max_stay=7,
    on_site_security=True,
    other_rules="other rules",
    other_services="other services",
    overall_rating=3,
    phone="2125551212",
    program_fees="program fees",
    room_styles_other="room styles other",
    shelter_programs_other="shelter programs other",
    shelter_types_other="shelter types other",
    subjective_review="subjective review",
    supervisorial_district=1,
    total_beds=10,
    website="https://shelter.example.com",
    accessibility=related(accessibility_recipe),
    cities=related(city_recipe),
    demographics=related(demographic_recipe),
)

from model_bakery import baker
from shelters.tests.shelter_recipes import shelter_recipe

class ShelterQueryTestCase(GraphQLTestCaseMixin, ParametrizedTestCase, TestCase):
    def setUp(self):
        super().setUp()
        self.maxDiff = None
        self.shelter_count = 2

        # Use recipe to create shelters with all related objects
        self.shelters = shelter_recipe.make(_quantity=self.shelter_count)

        # Add contacts directly using baker
        self.contacts = baker.make(
            "shelters.ContactInfo",
            shelter=self.shelters[0],
            _quantity=2,
            contact_name=seq("shelter contact "),
            contact_number=seq("212555121"),
        )

        # Add photos
        self._setup_images()

    def _setup_images(self):
        file_content = b"\x47\x49\x46\x38\x39\x61\x01\x00\x01\x00\x00\x00\x00\x21\xf9\x04\x01\x0a\x00\x01\x00\x2c\x00\x00\x00\x00\x01\x00\x01\x00\x00\x02\x02\x4c\x01\x00\x3b"
        file = SimpleUploadedFile(name="file.jpg", content=file_content)
        self.exterior_photo = baker.make("shelters.ExteriorPhoto", shelter=self.shelters[0], file=file)
        self.interior_photo = baker.make("shelters.InteriorPhoto", shelter=self.shelters[0], file=file)
```

Reply:
working on this in separate PR